code/src/App.js

```diff
@@ -1,9 +1,99 @@
-import React from 'react'
+/* eslint-disable */
+import React, { useState, useEffect } from 'react';
+
+import { Header } from './components/Header';
+import { MessageList } from './components/MessageList';
+import { MessageForm } from './components/MessageForm';
+import { THOUGHTS_URL } from './components/urls';
 
 export const App = () => {
+  // Stores message-input and changes the state of messages.
+  const [messages, setMessages] = useState([]);
+  // useEffect's functionality is called when component is mounted - after
+  // the return of jsx. useEffect allows us to catch a moment in the lifecycle of
+  // the App-component. useEffect calls the fetchMessages-function
+  useEffect(() => {
+    fetchMessages();
+  }, []);
+
+  // A funtion that fetches data (messages) from the server and stores
+  // the value using useState and setMessages.
+  const fetchMessages = () => {
+    fetch(THOUGHTS_URL)
+      .then((res) => res.json())
+      .then((data) => setMessages(data))
+      .catch((error) => console.error(error));
+  };
+  // A function for posting a new message TO the server. The fetch
+  // posts a new message (posts new data) to the server. And then the
+  // fetch calls the function fetchMessages which once again fetches new
+  // data (messages) FROM the server to App.js. App.js again stores this new message
+  // and passes the value to the MessageList component using the prop messageList.
+  const postSingleMessage = (newMessage) => {
+    fetch(THOUGHTS_URL,
+      {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ message: newMessage })
+      })
+      .then(() => fetchMessages())
+      .catch((error) => {
+        console.error(error);
+      })
+  };
+  // A function that updates the number of likes by posting to the server. Posting new data - heart likes - 
+  // via App.js
+  // Matches the messageId with heart like.
+  // After the post, a new GET request is activated via fetchMessage.
+  const postHearts = (messageId) => {
+    fetch(`https://happy-thoughts-technigo.herokuapp.com/thoughts/${messageId}/like`, {
+      method: 'POST',
+      body: '',
+      headers: { 'Content-Type': 'application/json' }
+    }).then(() => {
+      //onLiked(messageId);
+      fetchMessages();
+    });
+  };
+
+  // The function onLiked below updates the browser locally but does not POST
+  // hearts to the server. Can be used to avoid time lag - to show likes if the response
+  // from the fetch takes some time. (true?)
```

Comment:
I think you're on to something here – it's fairly common for applications to _assume_ the request will work out without problems, and update it locally while at the same time sending to the server and waiting for the response. If, for some reason, the response would come back with a "fail" or similar, apps usually post "Something went wrong when liking the post", or similar. Good thinking!